### PR TITLE
spack.relocate: further coverage for ELF related functions

### PR DESCRIPTION
**Pull request by alalazo**
This PR:

- [x] Increase coverage for all the remaining ELF related functions in `spack.relocate`
- [x] Improve docstrings and naming for the same functions

### CODE REVIEW

File: lib/spack/spack/relocate.py [hunk 1]
```diff
@@ -706,97 +710,143 @@ def make_macho_binaries_relative(cur_path_names, orig_path_names,
...
-    including the path the the spack sbang script
+    # TODO: reduce the number of arguments (8 seems too much)
+    sbang_regex = r'#!/bin/bash {0}/bin/sbang'.format(orig_spack)
+    new_sbang = r'#!/bin/bash {0}/bin/sbang'.format(new_spack)
+
+    for file in files:
+        _replace_prefix_text(file, orig_install_prefix, new_install_prefix)
+        for orig_dep_prefix, new_dep_prefix in new_prefixes.items():
+            _replace_prefix_text(file, orig_dep_prefix, new_dep_prefix)
+        _replace_prefix_text(file, orig_layout_root, new_layout_root)
```
Comment (becker33) [inline lib/spack/spack/relocate.py:814]: Why do replace the layout root for text files but not binary files? Seems like this should be the same for both.

File: lib/spack/spack/test/relocate.py [hunk 2]
```diff
@@ -279,38 +300,72 @@ def test_relocate_elf_binaries_absolute_paths(hello_world, tmpdir):
...
+    ])
+    new_binary = copy_binary(orig_binary)
+
+    # Check this call succeed
+    spack.relocate.relocate_text_bin(
+        [str(new_binary)],
+        str(orig_binary.dirpath()), str(new_binary.dirpath()),
+        spack.paths.spack_root, spack.paths.spack_root,
+        {str(orig_binary.dirpath()): str(new_binary.dirpath())}
+    )
```
Comment (becker33) [inline lib/spack/spack/test/relocate.py:375]: Shouldn't we actually test that the replacement succeeded? We could read it in as binary ('rb' mode) and check that it contains the new string and not the old string.

**Review by becker33 — CHANGES_REQUESTED** [review]
Two requests/questions

**Review by gartung — APPROVED** [review]

File: lib/spack/spack/relocate.py (hunk 1 above)
Comment (alalazo) [reply lib/spack/spack/relocate.py:814]: For what is worth in this PR I didn't change any logic, only added unit tests, docstrings and made naming consistent for ELF related functions. If it is fine with you I can add a TODO like I did above for the number of arguments, and work on that in a following PR (so not to mix the trivial refactor done here with a change in logic).

Also, maybe @gartung has more to say on the rationale.

File: lib/spack/spack/test/relocate.py (hunk 2 above)
Comment (alalazo) [reply lib/spack/spack/test/relocate.py:375]: Nice idea. I probably have to modify the `hello_world` fixture to contain one of these paths in the data segment, but I think this is absolutely doable.

Comment (alalazo) [reply lib/spack/spack/test/relocate.py:375]: Done in b9de3b2

**Review by becker33 — APPROVED** [review]

File: lib/spack/spack/relocate.py (hunk 1 above)
Comment (becker33) [reply lib/spack/spack/relocate.py:814]: @alalazo that works for me.

**Comment by alalazo** [comment]
Thanks @becker33 !

**Pull request merged**